pennylane/interfaces/execution.py

```diff
@@ -382,34 +396,47 @@ def cost_fn(params, x):
         cache = LRUCache(maxsize=cachesize)
         setattr(cache, "_persistent_cache", False)
 
-    batch_execute = set_shots(device, override_shots)(device.batch_execute)
+    if new_device_interface:
+        batch_execute = device.execute
+    else:
+        batch_execute = set_shots(device, override_shots)(device.batch_execute)
 
     if expand_fn == "device":
-        expand_fn = lambda tape: device.expand_fn(tape, max_expansion=max_expansion)
+        if new_device_interface:
 
-    if gradient_fn is None:
-        # don't unwrap if it's an interface device
-        if "passthru_interface" in device.capabilities() or device.short_name == "default.mixed":
-            return batch_fn(
-                qml.interfaces.cache_execute(
-                    batch_execute, cache, return_tuple=False, expand_fn=expand_fn
-                )(tapes)
-            )
-        unwrapped_tapes = tuple(qml.transforms.convert_to_numpy_parameters(t) for t in tapes)
-        res = qml.interfaces.cache_execute(
-            batch_execute, cache, return_tuple=False, expand_fn=expand_fn
-        )(unwrapped_tapes)
+            def expand_fn(tape):  # pylint: disable=function-redefined
+                """A blank expansion function since the new device handles expansion in preprocessing."""
+                return tape
 
-        return batch_fn(res)
+        else:
 
-    if gradient_fn == "backprop" or interface is None:
-        return batch_fn(
-            qml.interfaces.cache_execute(
-                batch_execute, cache, return_tuple=False, expand_fn=expand_fn
-            )(tapes)
+            def expand_fn(tape):  # pylint: disable=function-redefined
+                """A wrapper around the device ``expand_fn``."""
+                return device.expand_fn(tape, max_expansion=max_expansion)
+
+    if gradient_fn in {None, "backprop"} or interface is None:

```

Comment:
Why is gradient_fn None added?

Reply:
If you look at above line 390, this logical structure holds for `gradient_fn = None`.